### ISSUE TITLE
When running the Maven plugin tests, a lot of JVM instances stay running

### ISSUE THREAD

**Issue by gsmet** [quarkusio/quarkus#3576]
I noticed that because my laptop was becoming totally unresponsive when running the Maven plugin tests due to all the memory being consumed by these running JVMs.

@geoand could you confirm if you have the same issue.

Basically I did:
```
cd git/quarkus
git pull upstream master
mvn clean install -DskipTests -DskipITs
cd devtools/maven
mvn clean install
```

Then when the `RemoteDevMojoIT` starts being executed, I have a lot of JVMs started that are not properly stopped, whereas I would expect them to be stopped after each test.

I think it worked before because it's the first time I notice this issue.

**Comment by geoand** [comment]
@gsmet I will try it out tomorrow and let you know because I am traveling for the next few hours

**Comment by geoand** [comment]
@gsmet I followed your steps and couldn't reproduce the problem

**Comment by gwenneg** [comment]
I'll try on my computer.

**Comment by gsmet** [comment]
Basically, what I do is having a `top` running sorted by memory. And `java` instances keep piling up while the tests are running.

**Comment by geoand** [comment]
I pulled the latest master but still couldn't reproduce 😣

**Comment by gsmet** [comment]
FWIW, it's instances of `quarkus:remote-dev`. Let me try again.

**Comment by gsmet** [comment]
Running:
```
watch -n 2 'ps fauxwww | grep quarkus:remote-dev'
```
shows the problem very well.

I just reproduced it.

**Comment by geoand** [comment]
Trying again

**Comment by geoand** [comment]
Seems like I was able to reproduce this time around

**Comment by gsmet** [comment]
I think I found the culprit: https://github.com/quarkusio/quarkus/blob/master/devtools/maven/src/test/java/io/quarkus/maven/it/verifier/RunningInvoker.java#L55

We should also kill the `remote-dev` instances. I'm preparing a PR.

Note that it's a bit brutal IMHO but I suppose we can't do better for now.

**Comment by geoand** [comment]
I think it will be just fine :wink:
